gateway/main.go

```diff
@@ -181,7 +181,7 @@ func (g *Gateway) profileHandler(c *gin.Context) {
 		return
 	}
 
-	pth, err := ipfs.ResolveIPNS(g.Node.Ipfs(), rootId)
+	pth, err := ipfs.ResolveIPNS(g.Node.Ipfs(), rootId, time.Second * 30)
```

Comment:
Should the timeout also be a configuration option? Considering how slow IPNS can be.

Reply:
yeah, prob should be if we start using IPNS for anything that makes the gateway endpoint useful. the finding from this work was that it's still not going to work for anything we've go, so backing off it.